### PR TITLE
docs: add external-plugin and update README

### PR DESCRIPTION
**Pull request by okaybase**
### Pre-submission checklist:

* [x] Did you explain what problem does this PR solve? Or what new features have been added?
* [ ] Have you added corresponding test cases?
* [x] Have you modified the corresponding document?
* [x] Is this PR backward compatible? **If it is not backward compatible, please discuss on the [mailing list](https://github.com/apache/apisix/tree/master#community) first**

### CODE REVIEW

File: docs/zh/latest/external-plugin.md [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+
+注意：在 Mac 上，APISIX `v2.6` 无法管理该插件运行程序。
+
+在开发过程中，我们希望单独运行插件运行程序，这样就可以重新启动它，而无需先重新启动 APISIX 。
+
+通过指定环境变量 `APISIX_LISTEN_ADDRESS`, 我们可以使插件运行程序监听一个固定的地址。
+例如：
+
+```bash
+APISIX_LISTEN_ADDRESS=unix:/tmp/x.sock ./the_runner
```
Comment (tzssangglass) [inline docs/zh/latest/external-plugin.md:69]: what is `./the_runner`

File: docs/zh/latest/external-plugin.md [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+
+同时，你需要配置 APISIX 发送 RPC 请求到该固定的地址：
+
+```yaml
+ext-plugin:
+  # cmd: ["blah"] # don't configure the executable!
+  path_for_test: "/tmp/x.sock" # without 'unix:' prefix
+```
+
+在生产环境，不应该使用 `path_for_test` 并且 unix socket 路径将动态生产。
```
Comment (tzssangglass) [inline docs/zh/latest/external-plugin.md:82]:
```suggestion
在生产环境，不应该使用 `path_for_test`，并且 unix domain socket 的路径是动态生成的。
```

Comment (spacewander) [inline docs/zh/latest/external-plugin.md:82]:
```suggestion
在生产环境，不应该使用 `path_for_test`，此时监听的地址将动态生成。
```

File: docs/zh/latest/external-plugin.md [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+    - MY_ENV_VAR
+```
+
+### APISIX 使用 SIGKILL 终止插件运行程序，而不是使用 SIGTERM！
+
+自 `v2.7`，当跑在 OpenResty 1.19+ 时，APISIX 将使用 SIGTERM 来停止插件运行程序。
+
+但是，APISIX 需要等待插件运行程序退出，这样我们才能确保资源得以被释放。
+
+因此，我们先发送 SIGTERM 。然后在1秒后，如果插件运行程序仍然在运行，我们将发送 SIGKILL 。
```
Comment (spacewander) [inline docs/zh/latest/external-plugin.md:104]:
```suggestion
因此，我们先发送 SIGTERM 。然后在 1 秒后，如果插件运行程序仍然在运行，我们将发送 SIGKILL 。
```

File: docs/zh/latest/external-plugin.md [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+  path_for_test: "/tmp/x.sock" # without 'unix:' prefix
+```
+
+在生产环境，不应该使用 `path_for_test` 并且 unix socket 路径将动态生产。
+
+## 常见问题
+
+### 由 APISIX 管理时，插件运行程序无法访问我的环境变量
+
+自`v2.7`，APISIX 可以将环境传递给插件运行程序。
```
Comment (spacewander) [inline docs/zh/latest/external-plugin.md:88]:
```suggestion
自 `v2.7`，APISIX 可以将环境变量传递给插件运行程序。
```

File: docs/zh/latest/external-plugin.md [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+-->
+
+## 什么是外部插件和插件运行程序
+
+APISIX 支持使用 Lua 语言写插件，这种类型的插件在 APISIX 内部执行。
+有时候你想使用其他语言来开发插件，因此，APISIX 支持以边车的方式加载和运行你写的插件。
+这里的边车就叫做插件运行程序，你写的插件叫做外部插件。
```
Comment (spacewander) [inline docs/zh/latest/external-plugin.md:28]: We should keep the terms unchanged:
- External Plugin
- Plugin Runner
- Sidecar

File: docs/zh/latest/external-plugin.md (hunk 1 above)
Comment (okaybase) [reply docs/zh/latest/external-plugin.md:69]: done~ thanks~

File: docs/zh/latest/external-plugin.md (hunk 2 above)
Comment (okaybase) [reply docs/zh/latest/external-plugin.md]: okay~ done~ thanks~

File: docs/zh/latest/external-plugin.md (hunk 5 above)
Comment (okaybase) [reply docs/zh/latest/external-plugin.md]: okay~ done~ thanks~ @spacewander

File: docs/zh/latest/external-plugin.md [hunk 6]
```diff
@@ -0,0 +1,104 @@
...
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+-->
+
+## 什么是 `External Plugin` 和 `Plugin Runner`
```
Comment (spacewander) [inline docs/zh/latest/external-plugin.md:24]: We can remove the quote around, as APISIX doesn't have quotes around it.

**Review by spacewander — APPROVED** [review]

File: docs/zh/latest/external-plugin.md (hunk 6 above)
Comment (okaybase) [reply docs/zh/latest/external-plugin.md]: okay~ done~ thanks~

**Review by tzssangglass — APPROVED** [review]

**Pull request merged**